test/date_range_test.rb

```diff
@@ -0,0 +1,157 @@
+require_relative "test_helper"
+
+describe Hotel::DateRange do
+
+	describe "#initialize" do
+
+		before do
+			start_date = Date.new(2019, 12, 23)
+			end_date = Date.new(2019, 12, 28)
+
+			@new_range = Hotel::DateRange.new(start_date, end_date)
+		end
+
+		it "creates an instance of DateRange" do
+			expect(@new_range).must_be_instance_of Hotel::DateRange
+		end
+
+	end
+
+	describe "#include_range?" do
+		
+		before do
+			start_date = Date.new(2019, 12, 23)
+			end_date = Date.new(2020, 1, 28)
+
+			@new_range = Hotel::DateRange.new(start_date, end_date)
+		end
+
+		describe "when booking" do
```

Comment:
Consider adding tests that explicitly test for the all the different ways dates can and can't overlap

```
describe "overlap?" do
    before do
      start_date = Date.new(2017, 01, 01)
      end_date = start_date + 3

      @range = Hotel::DateRange.new(start_date, end_date)
    end

    it "returns true for the same range" do
      start_date = @range.start_date
      end_date = @range.end_date
      test_range = Hotel::DateRange.new(start_date, end_date)

      expect(@range.overlap?(test_range)).must_equal true
    end

    xit "returns true for a contained range" do
    end

    xit "returns true for a range that overlaps in front" do
    end

    xit "returns true for a range that overlaps in the back" do
    end

    xit "returns true for a containing range" do
    end

    xit "returns false for a range starting on the end_date date" do
    end

    xit "returns false for a range ending on the start_date date" do
    end

    xit "returns false for a range completely before" do
    end

    xit "returns false for a date completely after" do
    end
  end
```